snuba/subscriptions/scheduler.py

```diff
@@ -39,3 +42,47 @@ def find(self, interval: Interval[datetime]) -> Iterator[ScheduledTask[TTask]]:
         ascending order.
         """
         raise NotImplementedError
+
+
+# XXX: This is all temporary until we get these types properly introduced
+PartitionId = NewType("PartitionId", int)
+
+
+@dataclass(frozen=True)
+class SubscriptionIdentifier:
+    partition: PartitionId
+    uuid: UUID
+
+
+class Subscription(NamedTuple):
+    identifier: SubscriptionIdentifier
+    data: SubscriptionData
+
+
+class SubscriptionScheduler(Scheduler[Subscription]):
+    def __init__(self) -> None:
+        self.__subscriptions: Dict[UUID, Subscription] = {}
+
+    def set(self, value: Subscription) -> None:
+        self.__subscriptions[value.identifier.uuid] = value
+
+    def delete(self, key: UUID) -> None:
+        self.__subscriptions.pop(key, None)
+
+    def clear(self) -> None:
+        # XXX: Not part of the standard interface. Since we want to periodically clear
+        # the cache, whatever is managing this should probably clear and refetch
+        self.__subscriptions = {}
+
+    def find(
+        self, interval: Interval[datetime]
+    ) -> Iterator[ScheduledTask[Subscription]]:
+        for uuid, subscription in self.__subscriptions.items():
+            resolution = subscription.data.resolution.total_seconds()
+            for i in range(
+                math.ceil(interval.lower.timestamp() / resolution),
+                math.ceil(interval.upper.timestamp() / resolution),
+            ):
+                yield ScheduledTask(
+                    datetime.fromtimestamp(i * resolution), subscription,
+                )
```

Comment:
I forget why we decided this (I thought there was a specific reason), but I don't think this holds true wit this implementation: https://github.com/getsentry/snuba/blob/cf3f35cb050d50a513fe2e6f85be6f32de24f40c/snuba/subscriptions/scheduler.py#L38-L39

Reply:
I think the main thing is that we need to return things ordered by timestamp within a subscription. It doesn't matter much other than that imo. I might just update the docstring to include that?

Alternatively we can just switch the loops

Reply:
No strong opinion about this one, as long as the implementation and the docstring match I'm fine with updating whichever one is easier.

Reply:
I ended up switching things around. It actually makes the code a little cleaner, and means we don't have to iterate through every subscription on every tick.